network_plugin/public_nat.py

```diff
@@ -0,0 +1,109 @@
+from cloudify import ctx
+from cloudify import exceptions as cfy_exc
+from cloudify.decorators import operation
+from vcloud_plugin_common import with_vca_client, get_vcloud_config
+from network_plugin import check_ip, save_gateway_configuration, get_vm_ip, isExternalIpAssigned, CREATE, DELETE
+from network_plugin.network import VCLOUD_NETWORK_NAME
+
+PUBLIC_IP = 'public_ip'
+
+
+@operation
+@with_vca_client
+def net_connect_to_nat(vca_client, **kwargs):
+    prepare_network_operation(vca_client, CREATE)
+
+
+@operation
+@with_vca_client
+def net_disconnect_from_nat(vca_client, **kwargs):
+    prepare_network_operation(vca_client, DELETE)
+
+
+@operation
+@with_vca_client
+def server_connect_to_nat(vca_client, **kwargs):
+    prepare_vm_operation(vca_client, CREATE)
+
+
+@operation
+@with_vca_client
+def server_disconnect_from_nat(vca_client, **kwargs):
+    prepare_vm_operation(vca_client, DELETE)
+
+
+def prepare_network_operation(vca_client, operation):
+    try:
+        network_name = ctx.source.instance.runtime_properties[VCLOUD_NETWORK_NAME]
+        vdc_name = get_vcloud_config()['vdc']
+        public_ip = check_ip(ctx.target.node.properties['nat']['public_ip'])
```

Comment:
For floating ip we support public ip allocation from the pool of free ips. We should also support that for PublicNat node. So 'public_ip' property should be optional.